clients/src/main/java/org/apache/kafka/clients/admin/DeleteShareGroupsResult.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.clients.admin;
+
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.annotation.InterfaceStability;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * The result of the {@link Admin#deleteShareGroups(Collection <String>, DeleteShareGroupsOptions)} call.
+ * <p>
+ * The API of this class is evolving, see {@link Admin} for details.
+ */
+@InterfaceStability.Evolving
+public class DeleteShareGroupsResult extends DeleteConsumerGroupsResult {
+    DeleteShareGroupsResult(final Map<String, KafkaFuture<Void>> futures) {
+        super(futures);
+    }
+}
```

Comment:
The implementation for the public methods here could be added.

Reply:
As of now, it is exactly the same as the parent class.

Reply:
Yes, OK. I suggest making a common superclass `DeleteGroupsResult` and having both `DeleteConsumerGroupsResult` and `DeleteShareGroupsResult` inheriting from that.